### PR TITLE
LPS-152271 ClayIcons#FlagIconsCanRender

### PR DESCRIPTION
**Pull request by Esther-OC**
The goal of this PR is the automation of the test case ClayIcons#FlagIconsCanRender, to check that the flag icons are correctly displayed at the localization button.

Jira ticket: [LPS-152271](https://issues.liferay.com/browse/LPS-152271)

### CODE REVIEW

**Comment by Esther-OC** [comment]
Hi @john-co 
I solved the issue I had the previous day, you if you can review it whenever you can, I would appreciate it.  Thanks!

**Comment by john-co** [comment]
ci:test:sf

**Comment by john-co** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
Just started reviewing :)

**Comment by john-co** [comment]
Hi @Esther-OC , please remember to run the ci test suites when ready for review. Or attach the poshi report of the new test to the JIRA ticket. Reviewer needs to know that the poshi test passes somehow as noted in [FI QA Confluence](https://liferay.atlassian.net/wiki/spaces/QA/pages/1977319517/Frontend+Infra+QA+Onboarding#Poshi-Automation-Testing) .

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			Click(locator1 = "Icon#FLAG_enUS");
+		}
+
+		task ("Then flag icons are present on the button and dropdown menu items") {
+			if (IsElementPresent(locator1 = "Icon#FLAG_caES")) {
+				AssertTextEquals(
+					key_text = "catala_2d_espanya",
+					locator1 = "Language#ANY",
+					value1 = "català-Espanya");
+			}
```
[the remark this answers is not in the document]
Comment (john-co) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase:66]: If we're not using ocular, I think something along the lines of LanguagePortlet.viewLanguages macro under Icon and else if Icon Menu section would document better what we're trying to assert.

**Review by john-co — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase (hunk 1 above)
[the remark this answers is not in the document]
Comment (Esther-OC) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase]: As commented, we finally did not use Ocular.
We have solved it checking the combination of the flag icon with the country, by going over the tree and checking the "following sibling" of the flag.  If his sibling is found, it means that the country matches the flag, so here we have the assertion we are looking for ;-)

Language.path:

```
<td>FLAG_AND_LANGUAGE</td>
	<td>//*[name()='svg']//*[contains(@href,'${key_flagID}')]//..//..//following-sibling::span[contains(text(),'${key_languageID}')]</td>
```

**Comment by Esther-OC** [comment]
cc/ @manoelcyreno

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+		task ("Given language portlet") {
+			Navigator.gotoSitePage(
+				pageName = "Language Test Page Name",
+				siteName = "Test Site Name");
+		}
+
+		task ("When click on localization button") {
+			AssertElementPresent(
+				key_flagID = "en-us",
+				locator1 = "Icon#FLAG_ID");
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase:55]: Remove assertions from When steps. Only assertions should be found in Then steps.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+			LanguagePortlet.viewCurrentLanguage(locale = "en-US");
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase:57]: same

**Review by john-co — CHANGES_REQUESTED** [review]
Check the SF error, it's related to not having LPS-152271 prefix in some of your git commit messages.

**Comment by Esther-OC** [comment]
ci:test:sf

**Comment by Esther-OC** [comment]
ci:test:frontend:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase (hunk 2 above)
Comment (Esther-OC) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase]: Understood, thanks.
I did it as a way to verify that the language selector widget had the en-US flag.

**Comment by Esther-OC** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase (hunk 2 above)
Comment (john-co) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase]: One way to verify is to use VerifyElementPresent function or WaitForElementPresent.

Another way is to do an if else to help ensure if it's not en-US then change it to en-US.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase [hunk 4]
```diff
@@ -6,65 +6,95 @@ definition {
...
+
+
+	- task ("Then flag icons are present on the button and dropdown menu items") {
+	The task says we're asserting flag icons meaning multiple icons but this task is only asserting 1 icon.
+	Also, were you able to try out ocular to see if it would better meet this need for all icons rendering ok?
+	If we're not using ocular, I think something along the lines of LanguagePortlet.viewLanguages macro under Icon and else if Icon Menu section would document better what we're trying to assert
+
+	**/
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase]: I'm not comfortable about leaving this trail of review history in these intermediate commits.

**Review by john-co — COMMENTED** [review]
Thanks Esther! I squashed the fixes you made into one commit to remove the trail and also cleaned up the test. Please see https://github.com/liferay-frontend/liferay-portal/pull/2302 for the changes.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase (hunk 2 above)
Comment (john-co) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase]: LanguagePortlet.selectLanguagePG pretty much includes the if else

Comment (Esther-OC) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase]: Noted, thanks!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase (hunk 4 above)
Comment (Esther-OC) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase]: Yes, sure, you're totally right!  I didn't realize that it was saved in a commit, it was just the notes I needed when working on it but I thought they were just saved locally and I deleted them before making the commit.
Git is a snitch! 

Sorry for the inconveniences and thanks for letting me know, I will take into account for the future.

**Pull request closed without merging**